### DISCUSSION TITLE
How to wrap a function with a PyObject * return value?

### DISCUSSION THREAD

**Discussion by pemsley** [wjakob/nanobind#417] [Q&A] [answered]
I can't seem to find the right way to wrap a function with a PyObject * return value. I'm missing something simple I think.
Here is an example that illustrates my problem:

```
#include <nanobind/nanobind.h>
namespace nb = nanobind;

class Eclass {
public:
   Eclass() {}
   PyObject *get_value() { PyObject *r = PyFloat_FromDouble(1.1); return r; }
};

NB_MODULE(eclass, m) {

   nb::class_<Eclass>(m, "eclass")
      .def(nb::init<>())
      .def("get_value", &Eclass::get_value)
      ;

}
```

And in python3:

```
import eclass
e = eclass.eclass()
e.get_value()
```

I get:

> Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
    get_value(self) -> _object

Thanks.

**Comment by wjakob** [comment, marked as answer]
It's not supported at the moment. However, you can easily make your own type caster for it. There is a type caster for `nb::handle` which borrows instead of stealing the reference, a somewhat questionable choice in retrospect.

**Reply by pemsley** [reply]
OK, thank you.
(And thanks for your speedy reply.)
